File: skills/foundation-models/tests/run_all.py
#!/usr/bin/env python3
"""Master validation script for the foundation-models skill.

Validates the skill's central claim (HVG+PCA is a strong baseline) and the two
input contracts that fail silently (rank-value tokenization, gene vocabulary).

Usage:
    python run_all.py              # run all tests
    python run_all.py baseline     # run one test

Expected runtime: 2-5 minutes (30 MB download on first run, then cached)

Requirements:
    scanpy, scikit-learn, scipy

Optional:
    torch and a foundation model checkpoint. Model weights are multi-gigabyte,
    so the comparison against a real embedding is skipped when absent and the
    baseline is still established.
"""

import subprocess
import sys
from pathlib import Path

TESTS = ["vocabulary", "tokenization", "baseline"]  # cheapest first

args = sys.argv[1:]
if args:
    if args[0] not in TESTS:
        sys.exit(f"Unknown test: {args[0]}. Choose from: {', '.join(TESTS)}")
    tests = [args[0]]
else:
    tests = TESTS

try:
    from importlib.metadata import version

    sc_v = version("scanpy")
    sk_v = version("scikit-learn")
except Exception:
    sys.exit("scanpy and scikit-learn are required: pip install scanpy scikit-learn")

print("Foundation Models Skill Validation")
print("===================================")
print(f"Python:       {sys.version.split()[0]}")
print(f"scanpy:       {sc_v}")
print(f"scikit-learn: {sk_v}")

try:
    print(f"torch:        {version('torch')}")
except Exception:
    print("torch:        not installed (model comparison will be skipped)")

print(f"Tests to run: {', '.join(tests)}\n")

script_dir = Path(__file__).parent
failed = []

for test in tests:
    script = script_dir / f"validate_{test}.py"
    if not script.exists():
        print(f"Script not found: {script}")
        failed.append(test)
        continue
    print(f"\n--- Running {test} validation ---\n")
    result = subprocess.run([sys.executable, str(script)], check=False)
    if result.returncode != 0:
        failed.append(test)

print("\n===================================")
if failed:
    print(f"Failed: {', '.join(failed)}")
    sys.exit(1)
print("All validations passed.")

File: skills/drug-response/tests/run_all.R
#!/usr/bin/env Rscript
# Master validation script for the drug-response skill.
#
# Runs on simulated dose-response and prediction problems with known ground
# truth, so no PharmacoSet download (multi-gigabyte) is needed. The curve test
# additionally cross-checks against drc and nplr when they are installed.
#
# Usage:
#   Rscript run_all.R              # all tests
#   Rscript run_all.R curves       # one test
#
# Expected runtime: under 3 minutes
#
# Requirements:
#   curves      base R; optionally drc and nplr
#   prediction  glmnet

args <- commandArgs(trailingOnly = TRUE)

tests <- c("curves", "prediction")
if (length(args) > 0) {
  tests <- match.arg(args[1], tests)
}

cat("Drug Response Skill Validation\n")
cat("===============================\n")
cat(sprintf("Date: %s\n", Sys.time()))
cat(sprintf("R version: %s\n", R.version.string))

for (p in c("drc", "nplr", "glmnet")) {
  v <- tryCatch(as.character(packageVersion(p)), error = function(e) "not installed")
  cat(sprintf("%s: %s\n", p, v))
}
cat(sprintf("Tests to run: %s\n\n", paste(tests, collapse = ", ")))

script_dir <- tryCatch(dirname(sys.frame(1)$ofile), error = function(e) ".")
if (is.null(script_dir) || is.na(script_dir) || script_dir == "") script_dir <- "."

failed <- character()

for (test in tests) {
  script <- file.path(script_dir, sprintf("validate_%s.R", test))
  if (file.exists(script)) {
    cat(sprintf("\n--- Running %s validation ---\n\n", test))
    status <- system2("Rscript", script)
    if (!identical(status, 0L)) failed <- c(failed, test)
  } else {
    cat(sprintf("Script not found: %s\n", script))
    failed <- c(failed, test)
  }
}

cat("\n===============================\n")
if (length(failed) == 0) {
  cat("All validations passed.\n")
} else {
  cat(sprintf("Failed: %s\n", paste(failed, collapse = ", ")))
  quit(status = 1)
}
